src/client.rs

```diff
@@ -509,12 +487,14 @@ where
             //     Persist root metadata. The client MUST write the file to non-volatile storage as
             //     FILENAME.EXT (e.g. root.json).
 
-            self.store_metadata(&root_path, &MetadataVersion::None, &raw_signed_root)
-                .await;
+            self.local
+                .store_metadata(&root_path, &MetadataVersion::None, &raw_signed_root)
+                .await?;
```

Comment:
self.tuf was already updated with the new root metadata, so if this fails, then the caller takes some action to let this succeed, then calls update again, this loop won't fetch the root metadata again (because self.tuf already has it), resulting in the metadata never being stored locally.

Unfortunately, this ordering of operations aligns with the spec (5.1.6 and 5.1.7).  Maybe Client/Tuf should be tracking both a "trusted" root metadata and a "persisted" root metadata to ensure transient repository store errors don't lose metadata updates and eventually get persisted on retry?